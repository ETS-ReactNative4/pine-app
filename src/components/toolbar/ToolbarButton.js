import React, { Component } from 'react';
import { StyleSheet, View, Animated, TouchableWithoutFeedback } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  iconWrapper: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    width: 50,
    height: 50
  },
  icon: {
    position: 'absolute'
  }
});

export default class ToolbarButton extends Component {
  constructor() {
    super(...arguments);

    this._onPressIn = this._onPressIn.bind(this);
    this._onPressOut = this._onPressOut.bind(this);
  }

  UNSAFE_componentWillMount() {
    this._scale = new Animated.Value(1);
  }

  _onPressIn() {
    Animated.spring(this._scale, {
      toValue: 1.3,
      speed: 30,
      useNativeDriver: true
    }).start();
  }

  _onPressOut() {
    Animated.spring(this._scale, {
      toValue: 1,
      speed: 30,
      useNativeDriver: true
    }).start();
  }

  render() {
    const whiteOpacity = this.props.whiteOpacity || 0;
    const whiteIconStyle = { opacity: whiteOpacity };
    const defaultIconStyle = { opacity: 1 - whiteOpacity };

    const scaleStyle = {
      transform: [{ scale: this._scale }]
    };

    return (
      <View>
        <TouchableWithoutFeedback
          onPressIn={this._onPressIn}
          onPressOut={this._onPressOut}
          onPress={this.props.onPress}
        >
          <Animated.View style={[styles.iconWrapper, scaleStyle]}>
            <this.props.Icon style={[styles.icon, defaultIconStyle]} />
            <this.props.Icon style={[styles.icon, whiteIconStyle]} white={true} />
          </Animated.View>
        </TouchableWithoutFeedback>
      </View>
    );
  }
}

ToolbarButton.propTypes = {
  Icon: PropTypes.any.isRequired,
  onPress: PropTypes.func,
  whiteOpacity: PropTypes.number
};
