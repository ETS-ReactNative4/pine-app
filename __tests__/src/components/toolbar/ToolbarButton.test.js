import React from 'react';
import renderer from 'react-test-renderer';
import ToolbarButton from '../../../../src/components/toolbar/ToolbarButton';

describe('ToolbarButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <ToolbarButton Icon={'Icon'} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
