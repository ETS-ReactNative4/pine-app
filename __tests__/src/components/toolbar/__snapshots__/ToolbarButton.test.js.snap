// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToolbarButton renders correctly 1`] = `
<View>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 50,
        "justifyContent": "center",
        "transform": Array [
          Object {
            "scale": 1,
          },
        ],
        "width": 50,
      }
    }
  >
    <Icon
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "opacity": 1,
          },
        ]
      }
    />
    <Icon
      style={
        Array [
          Object {
            "position": "absolute",
          },
          Object {
            "opacity": 0,
          },
        ]
      }
      white={true}
    />
  </View>
</View>
`;
